conda/core/prefix_data.py

```diff
@@ -97,6 +142,80 @@ def __eq__(self, other: Any) -> bool:
             # neither prefix exists, raw comparison
             return self.prefix_path.resolve() == other.prefix_path.resolve()
 
+    def exists(self):
+        try:
+            return self.prefix_path.is_dir()
+        except OSError:
+            return False
+
+    def is_environment(self):
+        try:
+            return self._magic_file.is_file()
+        except OSError:
+            return False
+
+    def is_base(self):
+        return paths_equal(str(self.prefix_path), context.root_prefix)
+
+    @property
+    def is_writable(self):
+        if self.__is_writable == NULL:
+            if not self.is_environment():
+                is_writable = None
+            else:
+                is_writable = file_path_is_writable(self._magic_file)
+            self.__is_writable = is_writable
+        return self.__is_writable
+
+    def assert_exists(self):
+        if not self.exists():
+            raise EnvironmentLocationNotFound(self.prefix_path)
+
+    def assert_environment(self):
+        self.assert_exists()
+        if not self.is_environment():
+            raise DirectoryNotACondaEnvironmentError(self.prefix_path)
+
+    def assert_writable(self):
+        self.assert_environment()
+        if not file_path_is_writable(self._magic_file):
+            raise EnvironmentNotWritableError(self.prefix_path)
+
+    def validate_path(self, expand_path: bool = False) -> Path:
+        prefix_str = str(self.prefix_path)
+        if expand_path:
+            prefix_str = expand(prefix_str)
+            self.prefix_path = Path(prefix_str)
+
+        if os.pathsep in prefix_str:
+            raise CondaValueError(
+                f"Environment paths cannot contain '{os.pathsep}'. Prefix: '{prefix_str}'"
+            )
+
+        if " " in prefix_str:
+            log.warning(
+                "Environment paths should not contain spaces. Prefix: '%s'",
+                prefix_str,
+            )
+        parent = self.__class__(self.prefix_path.parent)
+        if parent.is_environment():
+            raise CondaValueError(
```

Comment:
Should environments ever be able to be nested? I can't tell if there's a legitimate use case for an environment to contain another environment at any level at all.

Reply:
Environments can be nested in this way:

```
$CONDA_ROOT
$CONDA_ROOT/envs/new-conda-env
$CONDA_ROOT/envs/new-conda-env/envs/another-conda-subenv
```

Not that we _should_ do that, but it's perfectly legal. We use this in the local dev setup, not sure if intentionally, but it does happen there.

What we are avoiding here is having this situation:

```
$CONDA_ROOT
$CONDA_ROOT/my-new-env
$CONDA_ROOT/envs/new-conda-env/another-conda-subenv
```

Reply:
This type of nesting, btw, is because `envs_dirs` includes the `base`-like (it has `conda` in it) environment path as one of the possible locations, as you have seen in your PRs. This is one of the reasons I want to remove that coupled behaviour.